### PR TITLE
driver can only have one active ride in state

### PR DESCRIPTION
**Pull request by Kwasi008**
- @jrmcgarvey 
- only driver can have one ride active
- updated test and fix broken test
- added new test for active ride code
-

### CODE REVIEW

File: app/controllers/api/v1/rides.rb [hunk 1]
```diff
@@ -162,6 +162,12 @@ class Rides < Grape::API
           status 404
           return {}
         end
+        #driver can only have one ride in a active state
+        if current_driver.rides.where( status: "picking-up" || "dropping-off" ||
```
Comment (kidatsy) [inline app/controllers/api/v1/rides.rb:166]: This is intended to be basically any status that isn't `scheduled` right? Could you just do `current_driver.rides.where.not( status: "scheduled" )`?

Comment (kidatsy) [reply app/controllers/api/v1/rides.rb:166]: or are there other statuses we're ignoring here, too?

Comment (jrmcgarvey) [reply app/controllers/api/v1/rides.rb:166]: KC, I think there are also pending and completed states -- I think the code is clearer as is.

**Review by jrmcgarvey — APPROVED** [review]
Looks Ok.

File: app/controllers/api/v1/rides.rb (hunk 1 above)
Comment (kidatsy) [reply app/controllers/api/v1/rides.rb:166]: Thanks for the clarification. My name is TC, by the way.

Comment (jrmcgarvey) [reply app/controllers/api/v1/rides.rb:166]: Oops, sorry for the typo.

**Pull request merged**